File: 1498. Number of Subsequences That Satisfy the Given Sum Condition/Solution.cpp
class Solution {
public:
    int numSubseq(vector<int>& nums, int target) {
        sort(begin(nums), end(nums));
        int res = 0, mod = 1000000007, left = 0, right = nums.size() - 1, p = 1;
        vector<int> power;
        for (int i = 0; i < nums.size(); i++) {
            power.push_back(p);
            p = (p << 1) % mod;
        }
        while (left <= right) {
            if (nums[left] + nums[right] > target) {
                right--;
            } else {
                res = (res + power[right - left]) % mod;
                left++;
            }
        }
        return res;
    }
};

File: 1498. Number of Subsequences That Satisfy the Given Sum Condition/Solution.py
class Solution:
    def numSubseq(self, nums: List[int], target: int) -> int:
        nums.sort()
        res, mod, left, right = 0, 1000000007, 0, len(nums) - 1
        power = [1 << i for i in range(len(nums))]
        while left <= right:
            if nums[left] + nums[right] > target:
                right -= 1
            else:
                res = (res + power[right - left]) % mod
                left += 1
        return res
